go-slang/src/vm/index.ts

```diff
@@ -42,14 +56,56 @@ export class GolangVM {
 
   run(instrs: VM.Instruction[]) {
     while (!(instrs[this.PC]._type === "DONE")) {
+      if (this.sleep_until !== undefined) {
+        if (new Date() < this.sleep_until) {
+          if (this.THREAD_QUEUE.length > 0) {
```

Comment:
code smell, not sure how to rearrange

a bit hard to do guard clauses here since we are in a while loop with a bunch of ifs, and there are still things we want to run downstairs